File: fastify/types/reagent.ts
export interface ReagentShowRequestQuery {
    name: string;
    page: number;
    pagesize: number;
}

export interface ReagentSearchParams {
    using: boolean;
    name?: {
        contains: string;
    };
}

export interface ReagentUpdateRequestBody {
    id: number;
    name: string;
    specifications: string;
    warn_number: number;
    price: number;
    storage_condition: string;
    warn_days: number;
    using: boolean;
    manufacturer: string;
    note: string;
}

export interface ReagentDelRequestBody {
    id: number;
}



export interface ReagentAddRequestBody {
    name: string;
    specifications: string;
    warn_number: number;
    price: number;
    storage_condition: string;
    warn_days: number;
    using: boolean;
    generate_lot: boolean;
    manufacturer: string;
    note: string;
} 

File: LabMangerAPI/LabMangerAPI/Models/Reagent.cs
using SqlSugar;

namespace LabMangerAPI.Models;

public class Reagent
{
    [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
    public int Id { get; set; } 
    
    [SugarColumn(Length = 100, IsNullable = false)]
    public string Name { get; set; } = "";
    
    [SugarColumn(Length = 100, IsNullable = false)]
    public string Specifications { get; set; } = "";

    [SugarColumn(IsNullable = false)]
    public int Price { get; set; } = 0;
    
    [SugarColumn(Length = 100, IsNullable = false)]
    public string StorageCondition { get; set; } = "";

    [SugarColumn(IsNullable = false)]
    public bool Active { get; set; } = true;
    
    [SugarColumn(Length = 100, IsNullable = false)]
    public string Manufacturer { get; set; } = "";
    
    [SugarColumn(Length = 200, IsNullable = false)]
    public string Note { get; set; } = "";

    [SugarColumn(IsNullable = false)]
    public int WarnNumber { get; set; } = 0;

    [SugarColumn(Length = 100, IsNullable = false)]
    public int WarnDays { get; set; } = 0;
    
    [SugarColumn(ColumnDataType = "datetime")]
    public DateTime CreateTime { get; set; } = DateTime.Now;
    
    [SugarColumn(IsNullable = false)]
    public int TeamId { get; set; } 
}
